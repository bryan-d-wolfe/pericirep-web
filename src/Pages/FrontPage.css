#FrontPage {
  display: flex;
  flex-direction: column;
  color: cyan;
}

form {
  display: flex;
  flex-direction: column;
  align-items: center;
}

main img {
  animation: FrontPage-logo-spin infinite 20s linear;
  height: 40vmin;
}
  
@keyframes FrontPage-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
  