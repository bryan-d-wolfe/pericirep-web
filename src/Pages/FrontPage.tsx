import React from 'react';
import './FrontPage.css';
import Main from '../components/Main';
import logo from '../logo.svg';

function FrontPage () {
  const text: string = "What's the magic word?";  
  return (
    <div id='FrontPage'>
      <Main title={text} imgSrc={logo} />
    </div>
  );
}

export default FrontPage;
