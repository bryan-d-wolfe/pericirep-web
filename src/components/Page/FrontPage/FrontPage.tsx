import React from "react";
import Main from "../../Main/Main";
import "./FrontPage";
import "./FrontPage.css";

function condition(text: string) {
  return text === "PeRiCiReP";
}

function FrontPage() {
  const title: string = "What's the magic word?";
  return (
    <div id="FrontPage">
      <Main title={title} condition={condition} />
    </div>
  );
}

export default FrontPage;
