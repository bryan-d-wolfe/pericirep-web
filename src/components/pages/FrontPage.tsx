import React from 'react';
import './FrontPage.css';
import Main from '../Main/Main';

function FrontPage () {
  const text: string = "What's the magic word?"; 
  return (
    <div id='FrontPage'>
      <Main title={text} />
    </div>
  );
}

export default FrontPage;
