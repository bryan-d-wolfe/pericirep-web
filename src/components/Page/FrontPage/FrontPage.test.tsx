import React from "react";
import snapshotTest from "../../../tests/snapshotTest";
import FrontPage from "./FrontPage";

test(
  "FrontPage renders correctly",
  () => snapshotTest(<FrontPage />),
);
