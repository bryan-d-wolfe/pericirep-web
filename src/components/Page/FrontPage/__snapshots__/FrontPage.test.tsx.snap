// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FrontPage renders correctly 1`] = `
<div
  id="FrontPage"
>
  <main>
    <h1>
      What's the magic word?
    </h1>
    <form>
      <textarea
        onChange={[Function]}
        value=""
      />
      <button
        onClick={[Function]}
      >
        Submit!
      </button>
    </form>
    <img
      src="joker.svg"
    />
  </main>
</div>
`;
